src/reference/asciidoc/amqp.adoc

```diff
@@ -906,6 +907,184 @@ public class AmqpJavaApplication {
+Requires a dedicated `RabbitTemplate` and a `CachingConnectionFactory` with the `publisherConfirms` property set to
+`true`. When a publisher confirm is received, and correlation data is supplied, it is written to either the
+confirm-ack-channel, or the confirm-nack-channel, depending on the confirmation type. The payload of the confirm is
+the correlation data as defined by this expression and the message will have a header 'amqp_publishConfirm' set to true
+(ack) or false (nack).
+For nacks, an additional header `amqp_publishConfirmNackCause` is provided.
+Examples: "headers['myCorrelationData']", "payload".
+If the expression resolves to a `Message<?>` instance (such as "`#this`"), the message
+emitted on the ack/nack channel is based on that message, with the additional header(s) added.
+_Optional_.
+
+<13> The channel to which positive (ack) publisher confirms are sent; payload is the correlation
+data defined by the _confirm-correlation-expression_.
+Requires the underlying `AsyncRabbitTemplate` to have its `enableConfirms` property set to true.
+_Optional, default=nullChannel_.
+
+<14> Since _version 4.2_. The channel to which negative (nack) publisher confirms are sent; payload is the correlation
+data defined by the _confirm-correlation-expression_.
+Requires the underlying `AsyncRabbitTemplate` to have its `enableConfirms` property set to true.
+_Optional, default=nullChannel_.
+
+<15> The channel to which returned messages are sent.
+When provided, the underlying amqp template is configured to return undeliverable messages to the gateway.
+The message will be constructed from the data received from amqp, with the following additional headers:
+_amqp_returnReplyCode, amqp_returnReplyText, amqp_returnExchange, amqp_returnRoutingKey_.
+Requires the underlying `AsyncRabbitTemplate` to have its `mandatory` property set to true.
+_Optional_.
+
+<16> When set to `false`, the endpoint will attempt to connect to the broker during application context initialization.
+This allows "fail fast" detection of bad configuration, by logging an error message if the broker is down.
+When true (default), the connection is established (if it doesn't already exist because some other component established
+it) when the first message is sent.
+
+
+[IMPORTANT]
+.RabbbitTemplate
+=====
+When using confirms and returns, it is recommended that the `RabbitTemplate` wired into the `AsyncRabbitTemplate` be
+dedicated.
+Otherwise, unexpected side-effects may be encountered.
+=====
+
+==== Configuring with Java Configuration
+
+The following configuration provides an example of configuring the outbound gateway using Java configuration:
+[source, java]
+----
+@Configuration
+public class AmqpAsyncConfig {
+
+    @Bean
+    @ServiceActivator(inputChannel = "amqpOutboundChannel")
+    public AmqpOutboundEndpoint amqpOutbound(AmqpTemplate asyncTemplate) {
+        AmqpAsyncOutboundGateway outbound = new AmqpOutboundEndpoint(asyncTemplate);
```

Comment:
The class name should be fixed to `AsyncAmqpOutboundGateway`.
For return type, too.